### DISCUSSION TITLE
CPlayerCameraDataNode - How u guys found functions of sync?

### DISCUSSION THREAD

**Discussion by SuriMax2** [YimMenu/HorseMenu#68] [Q&A]
Hi, question how u guys found the sync functions like the Health Spoof.
I remember that people can spoof spectate and position so im doing my best on this.

What i found so far is  CPlayerCameraDataNode
The camera locations: 

This values change while spectating some one, also when we out from spectate, back to normal:
{-1, 0, 0.xxxx}

```
class CPlayerCameraDataNode
{
public:
	char pad_0000[80]; //0x0000
	float SpectatePositionX; //0x0050
	float SpectatePositionY; //0x0054
	float SpectatePositionZ; //0x0058
	char pad_005C[996]; //0x005C
}; //Size: 0x0440
```

Im still on learn process im very new, i just start this year with c++ and i know is too early to start modding but i want to doe some stuffs, as example now that i found a class and some fields,  how i can found on ida functions that use this one, i alredy try to find a copy of what u guys doing on GTA5 but looks like is never called: 

Signature to what i found but i sure maybe is no correct one:
48 8B C4 48 89 58 08 48 89 68 18 56 57 41 56 48 83 EC 50 48 8B D9 0F 29 70 D8 48 8B CA 44 0F 29 40 ? 48 8B FA E8 ? ? ? ? 45 33 F6 84 C0 0F 85 ? ? ? ? 66 44 39 B3

**Comment by SuriMax2** [comment]
PD If this help some one that know fix crash:

There is a new crash:

The func .rdata:00007FF7F31B7B58                 dq offset sub_7FF7F1FB738
Looks like is inside of CPlayer Nodes (im no sure).
The break go into: L:369 
`for ( m = 0i64; (unsigned int)m < *(_DWORD *)(a2 + 36i64 * k_Iterator + 64); *(_DWORD *)v67 = 0 )`

Before this menu crash happen (i think is a chinese or fortitude), the stack show HandleCloneSync 4 functions before that function.

**Comment by Rxann** [comment]
> Hi, question how u guys found the sync functions like the Health Spoof.
> 
> I remember that people can spoof spectate and position so im doing my best on this.
> 
> 
> 
> What i found so far is  CPlayerCameraDataNode
> 
> The camera locations: 
> 
> 
> 
> This values change while spectating some one, also when we out from spectate, back to normal:
> 
> {-1, 0, 0.xxxx}
> 
> 
> 
> ```
> 
> class CPlayerCameraDataNode
> 
> {
> 
> public:
> 
> 	char pad_0000[80]; //0x0000
> 
> 	float SpectatePositionX; //0x0050
> 
> 	float SpectatePositionY; //0x0054
> 
> 	float SpectatePositionZ; //0x0058
> 
> 	char pad_005C[996]; //0x005C
> 
> }; //Size: 0x0440
> 
> ```
> 
> 
> 
> Im still on learn process im very new, i just start this year with c++ and i know is too early to start modding but i want to doe some stuffs, as example now that i found a class and some fields,  how i can found on ida functions that use this one, i alredy try to find a copy of what u guys doing on GTA5 but looks like is never called: 
> 
> 
> 
> Signature to what i found but i sure maybe is no correct one:
> 
> 48 8B C4 48 89 58 08 48 89 68 18 56 57 41 56 48 83 EC 50 48 8B D9 0F 29 70 D8 48 8B CA 44 0F 29 40 ? 48 8B FA E8 ? ? ? ? 45 33 F6 84 C0 0F 85 ? ? ? ? 66 44 39 B3
> 
> 
> 
> 
> 
> 

Did you succeed with this? I am looking to implement it if you cannot. I do not see anything about a camera node in rdr-classes. How did you find this in the first place, I am new to reversing and looking to reverse rdr2. Did you need to do any anti-vm work before starting to reverse the game?
